neural_modelling/src/neuron/synapses.c

```diff
@@ -99,26 +99,23 @@ static inline void print_synaptic_row(synaptic_row_t synaptic_row) {
     log_debug("----------------------------------------\n");
 
     // Get details of fixed region
-    address_t fixed_region_address = synapse_row_fixed_region(synaptic_row);
-    address_t fixed_synapses =
-            synapse_row_fixed_weight_controls(fixed_region_address);
-    size_t n_fixed_synapses =
-            synapse_row_num_fixed_synapses(fixed_region_address);
+    synapse_row_fixed_part_t *fixed_region =
+            synapse_row_fixed_region(synaptic_row);
+    address_t fixed_synapses = synapse_row_fixed_weight_controls(fixed_region);
+    size_t n_fixed_synapses = synapse_row_num_fixed_synapses(fixed_region);
     log_debug("Fixed region %u fixed synapses (%u plastic control words):\n",
-            n_fixed_synapses,
-            synapse_row_num_plastic_controls(fixed_region_address));
+            n_fixed_synapses, synapse_row_num_plastic_controls(fixed_region));
 
     for (uint32_t i = 0; i < n_fixed_synapses; i++) {
         uint32_t synapse = fixed_synapses[i];
         uint32_t synapse_type = synapse_row_sparse_type(
                 synapse, synapse_index_bits, synapse_type_mask);
 
-        log_debug("%08x [%3d: (w: %5u (=",
+        io_printf(IO_BUF, "%08x [%3d: (w: %5u (=",
```

Comment:
Any reason this is elevated?

Reply:
We don't want the per-line prefixing and suffixing that comes with `log_debug()` as it takes multiple prints to generate each line. It's all guarded by `#if LOG_LEVEL >= LOG_DEBUG` too, and fortunately (because of the issues with race conditions on IOBUF access) is almost never enabled.

I agree that it's messy.

Reply:
OK, missed the guarding, so OK.